doc/vrrp/VRRP_Adaptation_HLD.md

```diff
@@ -0,0 +1,712 @@
+# Virtual Router Redundancy Protocol Adaptation HLD #
+
+## Table of Content 
+
+- [Virtual Router Redundancy Protocol Adaptation HLD](#virtual-router-redundancy-protocol-adaptation-hld)
+  - [Table of Content](#table-of-content)
+    - [Revision](#revision)
+    - [Scope](#scope)
+    - [Definitions/Abbreviations](#definitionsabbreviations)
+    - [Overview](#overview)
+	  - [Requirements](#requirements)
+      - [Functional Requirements](#functional-requirements)
+	  - [Feature Description](#feature-description)
+      - [Functional Description](#functional-description)
+        - [Protocol version selection](#protocol-version-selection)
+        - [Pre-emption](#pre-emption)
+        - [Virtual Router Owner](#virtual-router-owner)
+        - [Virtual MAC Address](#virtual-mac-address)
+        - [VRRP Advertisement Frame](#vrrp-advertisement-frame)
+        - [Uplink Interface Tracking](#uplink-interface-tracking)
+    - [Architecture Design](#architecture-design)
+    - [High-Level Design](#high-level-design)
+      - [Operating environment](#operating-environment)
+      - [Scalability Requirements](#scalability-requirements)
+      - [Container](#container)
+        - [BGP container](#bgp-container)
+        - [SWSS container](#swss-container)
+      - [CoPP Configurations](#copp-configurations)
+      - [DB changes](#db-changes)
+        - [CONFIG_DB changes](#config_db-changes)
+        - [APPL_DB changes](#appl_db-changes)
+        - [ASIC_DB changes](#asic_db-changes)
+      - [Modules Design and Flows](#modules-design-and-flows)
+        - [ADD/DEL VRRP instance](#add/del-vrrp-instance)
+        - [Tracking interface](#tracking-interface)
+    - [SAI API](#sai-api)
+    - [Configuration and management](#configuration-and-management)
+      - [Manifest](#manifest)
+      - [CLI/YANG Model Enhancements](#cliyang-model-enhancements)
+        - [SNOiC VRRP YANG MODEL](#sonic-vrrp-yang-model)
+		- [CLI](#cli)
+          - [Config commands](#config-commands)
+          - [Show commands](#show-commands)
+    - [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+    - [Restrictions/Limitations](#restrictionslimitations)
+    - [Testing Requirements/Design](#testing-requirementsdesign)
+      - [Unit Test cases](#unit-test-cases)
+    - [Open/Action items](#openaction-items)
+ 
+### Revision  
+|  Rev  |  Date           |  Author     | Change Description |
+| :---  | :-------------- | :---------- | :----------------  |
+|  0.1  |  Aug-16-2023    | Philo-micas | Initial version    |
+
+### Scope  
+
+This document describes the high level design of frr-vrrpd adaptation to SONiC.
```

Comment:
Can you please capture alternative design with keepalived for VRRP? why is it important to choose FRR based VRRP rather keepalived based VRRP?

Reply:
yea, Keepalived and FRR are both routing software, and since FRR has already been choosed as the router protocol by SONiC long time ago, support muiltiple protocols except vrrp, such as BGP, OSPF, RIP, IS-IS, PIM, LDP, BFD,etc, 
	so I think no matter from the architecture, extension and aspect of opensource, and iteration speed, FRR will be more powerfull, and what more, above components have already be compiled in SONiC, just need to adapt and enable them.